spec/services/solidus_square/create_payment_service_spec.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+RSpec.describe SolidusSquare::CreatePaymentService do
+  subject(:handler) { described_class.new(source_id: "nonce", order: order, payment_method_id: payment_method.id) }
+
+  let(:options) {
+    { access_token: ENV['SQUARE_ACCESS_TOKEN'] || "abcde", environment: 'sandbox',
+      location_id: ENV['SQUARE_LOCATION_ID'] || 'location' }
+  }
+  let!(:payment_method) { create(:square_payment_method) }
+  let(:gateway) { SolidusSquare::Gateway.new(options) }
+  let!(:order) { create(:order_ready_to_complete, number: "R919717664", state: 'delivery', payment_state: nil) }
+  let(:status) { "CAPTURED" }
+  let(:square_response) do
+    {
+      amount_money: {
+        amount: order.total
+      },
+      card_details: {
+        status: status,
+        card: {
+          card_brand: "MASTERCARD",
+          # rubocop:disable Naming/VariableNumber
+          last_4: "9029",
+          # rubocop:enable Naming/VariableNumber
+          exp_month: 11,
+          exp_year: 2022,
+          card_type: "CREDIT"
+        },
+        avs_status: "AVS_ACCEPTED",
+      },
+      order_id: 12,
+      version_token: 12_345
+    }
+  end
+
+  before do
+    allow(Spree::PaymentMethod).to receive(:find).and_return(payment_method)
+    allow(gateway).to receive(:create_payment).and_return(square_response)
```

Comment:
You have missed the test for the parameters, and the test for this method:

https://github.com/nebulab/solidus_square/pull/49/commits/a3d3c036591ded5111759fa62cf01d55057cedf6#diff-00a6869703c8a39f974141b47b11dad1cb5940285bc866d9516e23e8dcb83495R57-R59

Reply:
https://github.com/nebulab/solidus_square/pull/49/commits/4920d74201d57ea1ec93f607f0f37f2a740c634a